airflow/providers/ftp/hooks/ftp.py

```diff
@@ -214,8 +219,10 @@ def store_file(
         else:
             input_handle = local_full_path_or_buffer
         remote_path, remote_file_name = os.path.split(remote_full_path)
+        current_path = conn.pwd()
         conn.cwd(remote_path)
         conn.storbinary(f"STOR {remote_file_name}", input_handle, block_size)
```

Comment:
and here you can provide the path instead of the name too

Reply:
Commit with your suggestion: 
https://github.com/apache/airflow/pull/35105/commits/86f92cabba9156bbe1dbce1aa46bc33fce901e27